Mage.Sets/src/mage/cards/s/SmugglersShare.java

```diff
@@ -0,0 +1,131 @@
+package mage.cards.s;
+
+import java.util.List;
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfEndStepTriggeredAbility;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.TargetController;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.TreasureToken;
+import mage.players.Player;
+import mage.watchers.common.CardsAmountDrawnThisTurnWatcher;
+import mage.watchers.common.PermanentsEnteredBattlefieldWatcher;
+
+/**
+ * One way or another, everyone gets paid what they're owed.
+ * Based on Gadrak, the Crown Scourge, Hoard Hauler, Lavaball Trap, Walker Of The Grove, Widespread Thieving, and Wound Reflection
+ *
+ * @author Slanman3755
+ */
+public final class SmugglersShare extends CardImpl {
+
+    public SmugglersShare(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{W}");
+
+        // At the beginning of each end step, draw a card for each opponent who drew two or more cards this turn, then
+        // create a Treasure token for each opponent who had two or more lands enter the battlefield under their control
+        // this turn.
+        Ability ability = new BeginningOfEndStepTriggeredAbility(new DrawCardSourceControllerEffect(SmugglersShareDrawValue.instance), TargetController.EACH_PLAYER, false);
+        ability.addEffect((new CreateTokenEffect(new TreasureToken(), SmugglersShareTreasureValue.instance)).concatBy(", then"));
+        ability.addWatcher(new CardsAmountDrawnThisTurnWatcher());
+        ability.addWatcher(new PermanentsEnteredBattlefieldWatcher());
+        this.addAbility(ability);
+    }
+
+    private SmugglersShare(final SmugglersShare card) {
+        super(card);
+    }
+
+    @Override
+    public SmugglersShare copy() {
+        return new SmugglersShare(this);
+    }
+}
+
+enum SmugglersShareDrawValue implements DynamicValue {
+    instance;
+
+    @Override
+    public int calculate(Game game, Ability source, Effect effect) {
+        Player controller = game.getPlayer(source.getControllerId());
```

Comment:
Need to check if controller is null. Otherwise it'll throw a NPE when you get to the for loop

Reply:
Can a card not have a controller? I got this check from Lavaball Trap I believe which also doesn't make this check,